src/devices/SenseHat/SenseHatLedMatrixI2c.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Diagnostics;
+using System.Drawing;
+using System.Runtime.CompilerServices;
+using System.Device.I2c.Drivers;
+
+namespace Iot.Device.SenseHat
+{
+    public class SenseHatLedMatrixI2c : SenseHatLedMatrix
+    {
+        public const int I2cAddress = 0x46;
+        private const int PixelLength = 3;
+        private const int FrameBufferLength = PixelLength * NumberOfPixelsPerRow * NumberOfPixelsPerColumn;
+        private const int ROffset = 0;
+        private const int GOffset = 8;
+        private const int BOffset = 16;
+        private I2cDevice _i2c;
+
+        public SenseHatLedMatrixI2c(I2cDevice i2cDevice = null)
+        {
+            _i2c = i2cDevice ?? CreateDefaultI2cDevice();
+            Clear(Color.Black);
+        }
+
+        public override void Write(ReadOnlySpan<Color> colors)
+        {
+            if (colors.Length != NumberOfPixels)
+                throw new ArgumentException($"`{nameof(colors)}` must have exactly {NumberOfPixels} elements.");
+
+            Span<byte> buffer = stackalloc byte[FrameBufferLength + 1];
+
+            // Register address of first pixel
+            buffer[0] = 0;
+
+            Span<byte> frameBuffer = buffer.Slice(1);
+
+            for (int i = 0; i < NumberOfPixels; i++)
+            {
+                (int x, int y) = IndexToPosition(i);
+                int address = PositionToAddress(x, y);
+                (byte r, byte g, byte b) = DestructColor(colors[i]);
```

Comment:
`Debug.Assert` that `colors.Length <= NumberOfPixels`

Reply:
there is a `(colors.Length != NumberOfPixels)` in the very begining of the function

Reply:
Sorry, missed that part.